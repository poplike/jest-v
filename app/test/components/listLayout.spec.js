import { shallowMount, createLocalVue } from '@vue/test-utils';
// import { render, fireEvent } from '@testing-library/vue'  UI 测试库
import '@testing-library/jest-dom';
import ListLayout from '@/components/listLayout.vue';
import ElementUI from 'element-ui';

// 创建临时Vue实例，挂载组件中使用的插件
const localVue = createLocalVue();
localVue.use(ElementUI);

localVue.prototype.$utils = {
  handleDate: (data) => {
    console.log(this, '-------------this');
    let result = '';
    if (date) {
      result = new Date(+new Date(date) + 8 * 3600 * 1000).toISOString();
      result = result.replace('Z', '');
    }
    return result;
  },
};

describe('ListLayout', () => {
  let wrapper, currentTimer, initFilterData;

  beforeEach(() => {
    currentTimer = new Date();
    initFilterData = {
      timerRange: [currentTimer, currentTimer],
      startTime: currentTimer,
      endTime: currentTimer,
      PageIndex: 1,
      PageSize: 10,
    };
    wrapper = shallowMount(ListLayout, {
      localVue,
      propsData: {
        url: 'url',
        filterData: {
          timerRange: [currentTimer, currentTimer],
          PageIndex: 1,
          PageSize: 10,
        },
        hasForm: true,
        dateRanges: [
          {
            rangeName: 'timerRange',
            childName: ['startTime', 'endTime'],
          },
        ],
      },
    });
  });
  afterEach(() => {
    wrapper.destroy();
  });

  // it('props_url', () => {
  //   expect(wrapper.props().url).toEqual('url');
  // });

  it('init_filterData', () => {
    expect(wrapper.vm.filterData).toEqual(initFilterData);
  });

  it('props_hasForm', async () => {
    wrapper.setProps({ hasForm: false });
    await localVue.nextTick();
    expect(wrapper.find('.form-wrap').element).toBeUndefined();
  });

  it('Snapshot', () => {
    expect(wrapper.vm.$el).toMatchSnapshot();
  });

  it('doFilter_btn', () => {
    let filterBtn = wrapper.find('el-button-stub');
    console.log(filterBtn);
    filterBtn.trigger('click');
    expect(wrapper.vm.filterData).toEqual(initFilterData);
  });
});
