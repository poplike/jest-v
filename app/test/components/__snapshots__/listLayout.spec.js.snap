// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListLayout Snapshot 1`] = `
<div>
  <div
    class="form-wrap"
  >
    <div
      class="list-layout-header"
    >
      <el-form-stub
        inline="true"
        labelsuffix=""
        labelwidth="100px"
        model="[object Object]"
        showmessage="true"
        validateonrulechange="true"
      >
         
        <el-form-item-stub
          inlinemessage=""
          showmessage="true"
        >
          <el-button-stub
            icon=""
            nativetype="button"
            type="primary"
          >
            搜索
          </el-button-stub>
        </el-form-item-stub>
      </el-form-stub>
       
      <!---->
    </div>
     
    <hr
      class="divider"
    />
  </div>
   
  <div
    class="table-wrap"
  />
   
  <div
    class="row-flex"
  >
    <el-pagination-stub
      background="true"
      class="mart-15 text-right row-flex-eq"
      currentpage="1"
      layout="total, sizes, prev, pager, next, jumper"
      pagercount="7"
      pagesize="10"
      pagesizes="10,20,30,40"
      total="0"
    />
     
  </div>
</div>
`;
